src/test/java/org/opensearch/neuralsearch/query/NeuralQueryBuilderTests.java

```diff
@@ -183,7 +184,11 @@ public void testFromXContent_withRescoreContext_thenBuildSuccessfully() {
         assertEquals(QUERY_TEXT, neuralQueryBuilder.queryText());
         assertEquals(MODEL_ID, neuralQueryBuilder.modelId());
         assertEquals(K, neuralQueryBuilder.k());
-        assertEquals(RescoreContext.getDefault(), neuralQueryBuilder.rescoreContext());
+        assertEquals(
+            RescoreContext.getDefault().getOversampleFactor(),
+            neuralQueryBuilder.rescoreContext().getOversampleFactor(),
+            DELTA_FOR_FLOATS_ASSERTION
+        );
```

Comment:
Since you are already using big decimal, please remove the delta here

Reply:
not sure what do you mean, assert requires third parameter in case we're comparing floats, and both arguments are float

Reply:
I mean you are using big decimal in the test `rrfNorm` method. You can be more strict, and the delta can be set to 0.